File: python/gcf.py
def greatest_common_factor(a, b):
    while b > 0:
        a, b = b, abs(a - b)
    return a


def main(args):
    args = args[1:]
    if len(args) == 0:
        print("Missing a and b! Try again with a and b provided!")
        return
    elif len(args) == 1:
        print("Missing b value! Try again!")
        return
    print(
        f"Greatest common factor between {args[0]} and {args[1]} is: {greatest_common_factor(int(args[0]), int(args[1]))}"
    )

File: c/gcf.c
#include <stdio.h>
#include <stdlib.h>

int greatest_common_factor(int a, int b){
  while(b > 0){
    int c = a;
    a = b;
    b = abs(c - b);
  }
  return a;
}

int run_gcf(int argc, char *argv[]){
  if(argc < 4){
    printf("Missing parameters for a and b, try again");
    return 1;
  }
  printf("%d\n", greatest_common_factor(atoi(argv[2]), atoi(argv[3])));
  return 0;
}